### PR TITLE
Update Kerberos library to latest major

### PR DESCRIPTION
**Pull request by kvch**
This PR updates the Kerberos library to the latest, v8.4.2. The version used in Sarama ATM is no longer supported. All fixes and features are added to the latest major.

New import path of the lib: `github.com/jcmturner/gokrb5/v8`

### CODE REVIEW

**Comment by bai** [comment]
Thanks for contributing!

LGTM, but I'll wait for another review for safety.

**Review by bai — APPROVED** [review]

File: go.mod
Comment (d1egoaz) [inline go.mod:16]: 🤔 why this version was downgraded from `v1.11.4`?

File: go.mod
Comment (d1egoaz) [inline go.mod:13]: this was also downgraded

**Comment by bai** [comment]
@kvch do you think you could rebase your branch and fix the downgrade issues in go.mod/go.sum? We're about to cut a new release soon.

File: go.mod
Comment (bai) [reply go.mod]: Fixed 🙏

File: go.mod
Comment (bai) [reply go.mod]: Fixed 🙏

**Review by d1egoaz — APPROVED** [review]

File: go.mod
Comment (d1egoaz) [reply go.mod]: thanks @bai

**Pull request merged**
